src/ProgramVariable.cpp

```diff
@@ -4,20 +4,26 @@
 ProgramVariable::ProgramVariable(Value *value) {
   this->value = value;
   this->rawName = dataflow::variable(value);
-  this->cleanedName = dataflow::variable(value);
+  this->cleanedName = this->rawName;
+  this->varIsIdentifier = true;
 
   if (this->cleanedName[0] == '%' || this->cleanedName[0] == '@') {
     this->cleanedName.erase(0, 1);
+  } else {
+    this->varIsIdentifier = false;
```

Comment:
Since this code appears twice (here and at line 26 below), it makes sense to put it in a separate function (since it's more than a single line now).